cmd/gen-go-sample/main.go

```diff
@@ -485,6 +484,29 @@ func (g *generator) paging(meth *descriptor.MethodDescriptorProto, pf *descripto
 	return err
 }
 
+func (g *generator) lro(meth *descriptor.MethodDescriptorProto, methConf GAPICMethod, valSet SampleValueSet) error {
+	p := g.pt.Printf
+
+	p("op, err := c.%s(ctx, req)", meth.GetName())
+	p("if err != nil {")
+	p("  return err")
+	p("}")
+	p("resp, err := op.Wait(ctx)")
+	p("if err != nil {")
+	p("  return err")
+	p("}")
+
+	lroConf := methConf.LongRunning
+	var typ initType
```

Comment:
Nit: Personally, I like to handle a condition (i.e. errors) directly and have happy path be the "fall through". It's easier to read imo.
```go
if lroConf.ReturnType == "" {
  return errors.E(nil, "LRO return type not given")
}

typ := initType{desc: g.descInfo.Type["."+lroConf.ReturnType]}
```

Reply:
SG